goalie/mesh_seq.py

```diff
@@ -142,27 +192,44 @@ def set_meshes(self, meshes):
                     f"{i}: {nc:7d} cells, {nv:7d} vertices,   max aspect ratio {mar:.2f}"
                 )
             debug(100 * "-")
+        self._time = [
+            ffunc.Function(ffs.FunctionSpace(mesh, "R", 0)) for mesh in meshes
+        ]
 
-    def plot(
-        self, **kwargs
-    ) -> Tuple[matplotlib.figure.Figure, matplotlib.axes._axes.Axes]:
+    def get_time(self, subinterval):
+        """
+        Get the time :class:`~.Function` associated with a given subinterval,
+        initialised to the value at the start of the subinterval.
```

Comment:
Connected to https://github.com/mesh-adaptation/goalie/issues/141#issuecomment-2066151474, this feels quite unintuitive to me - especially since it then requires doing `time += dt` (e.g. Line 165 in `bubble_shear.py`) before we actually solve the problem. This also goes against what firedrake demos are doing, where they do `time += dt` after the solve. I think it would be nice to keep it consistent.

I am not insisting on this, but I think I would prefer to say something like "... initialised to the time value associated with the first timestep of the subinterval" and below do `self._time[subinterval].assign(start_time + dt)`. 

I would just like to make a clear distinction about what is the first time that is being solved for in the subinterval: which is `start_time + dt`.

Reply:
Fair enough. Fixed in 5d297f2.